stl/inc/chrono

```diff
@@ -3385,9 +3385,10 @@ namespace chrono {
                                  | (_Can_rep_day ? _F_doy : 0)};
 
             const auto _Used{_Used_fields()};
-            const auto _Time_out_of_range{
-                _For_time_point
-                && (_Out_of_range(_Second, 0, 60) || _Out_of_range(_Minute, 0, 59) || _Out_of_range(_Hour_24, 0, 23))};
+            const auto _Max_seconds{(_Hour_24 == 23 && _Minute == 59) ? 60 : 59};

```

Comment:
This is incorrect when time zone offset is non-zero.
```cpp
istringstream{"2017-01-01T05:29:60+05:30"} >> parse("%FT%T%Ez"s, t)
```

Reply:
You're right, thanks. And now that I look at the logic, applying the UTC offset shouldn't be restricted to `time_points` that can represent minutes. It ought to work when both the offset and the `time_point` are hours.

I think your #2705 is likely to be more promising, so I'll close this one for now and make a separate PR for the offset issue.